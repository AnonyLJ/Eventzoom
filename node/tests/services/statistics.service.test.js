import logModel from '../../src/models/log.model';
import eventModel from '../../src/models/event.model';
import statisticsService from '../../src/services/statistics.service';

// Mock save. Mongoose save returns the Id of the document saved
const mockSave = jest.fn();
jest.mock('../../src/models/log.model', () => jest.fn().mockImplementation(() => ({ save: mockSave })));
jest.mock('../../src/models/event.model', () => jest.fn().mockImplementation(() => ({ save: mockSave })));


describe('testing logOccurence', () => {
	it('should run successfully', async () => {
		logModel.aggregate = jest.fn().mockImplementation(async () => 'stuff');
		eventModel.aggregate = jest.fn().mockImplementation(async () => 'stuff1');
		expect(await statisticsService.getStatistics()).toMatchSnapshot();
		expect(logModel.aggregate.mock.calls).toMatchSnapshot();
		expect(eventModel.aggregate.mock.calls).toMatchSnapshot();
	});
});
