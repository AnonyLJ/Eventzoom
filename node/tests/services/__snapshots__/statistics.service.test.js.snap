// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing logOccurence should run successfully 1`] = `
Array [
  Object {
    "name": "Most searched terms",
    "results": "stuff",
  },
  Object {
    "name": "Most viewed events",
    "results": "stuff",
  },
  Object {
    "name": "Events with most attendees",
    "results": "stuff1",
  },
]
`;

exports[`testing logOccurence should run successfully 2`] = `
Array [
  Array [
    Array [
      Object {
        "$match": Object {
          "event": "search",
        },
      },
      Object {
        "$group": Object {
          "_id": "$data.term",
          "count": Object {
            "$sum": 1,
          },
        },
      },
      Object {
        "$sort": Object {
          "count": -1,
        },
      },
      Object {
        "$limit": 5,
      },
      Object {
        "$project": Object {
          "count": "$count",
          "title": "$_id",
        },
      },
    ],
  ],
  Array [
    Array [
      Object {
        "$match": Object {
          "event": "visit",
        },
      },
      Object {
        "$group": Object {
          "_id": "$referencedEvent",
          "count": Object {
            "$sum": 1,
          },
        },
      },
      Object {
        "$sort": Object {
          "count": -1,
        },
      },
      Object {
        "$lookup": Object {
          "as": "event",
          "foreignField": "_id",
          "from": "events",
          "localField": "_id",
        },
      },
      Object {
        "$project": Object {
          "count": "$count",
          "title": Object {
            "$arrayElemAt": Array [
              "$event.title",
              0,
            ],
          },
        },
      },
      Object {
        "$limit": 5,
      },
    ],
  ],
]
`;

exports[`testing logOccurence should run successfully 3`] = `
Array [
  Array [
    Array [
      Object {
        "$unwind": "$attendees",
      },
      Object {
        "$group": Object {
          "_id": Object {
            "_id": "$_id",
            "title": "$title",
          },
          "count": Object {
            "$sum": 1,
          },
        },
      },
      Object {
        "$sort": Object {
          "count": -1,
        },
      },
      Object {
        "$project": Object {
          "_id": "$_id._id",
          "count": "$count",
          "title": "$_id.title",
        },
      },
      Object {
        "$limit": 5,
      },
    ],
  ],
]
`;
